src/features/copilotTools.ts

```diff
@@ -36,55 +50,181 @@ export class GetPackagesTool implements LanguageModelTool<IGetActiveFile> {
             deferredReturn.resolve({ content: [new LanguageModelTextPart(errorMessage)] } as LanguageModelToolResult);
         });
 
-        const parameters: IGetActiveFile = options.input;
+        const parameters: IResourceReference = options.input;
 
-        if (parameters.filePath === undefined || parameters.filePath === '') {
-            throw new Error('Invalid input: filePath is required');
+        if (parameters.resourcePath === undefined || parameters.resourcePath === '') {
+            throw new Error('Invalid input: resourcePath is required');
         }
-        const fileUri = Uri.file(parameters.filePath);
+        const resourcePath: Uri = Uri.file(parameters.resourcePath);
 
         try {
-            const environment = await this.api.getEnvironment(fileUri);
+            // environment info set to default values
+            const envInfo: EnvironmentInfo = {
+                type: 'no type found',
+                version: 'no version found',
+                packages: 'no packages found',
+                runCommand: 'no run command found',
+            };
+
+            // environment
+            const environment: PythonEnvironment | undefined = await this.api.getEnvironment(resourcePath);
             if (!environment) {
                 // Check if the file is a notebook or a notebook cell to throw specific error messages.
-                if (fileUri.fsPath.endsWith('.ipynb') || fileUri.fsPath.includes('.ipynb#')) {
+                if (resourcePath.fsPath.endsWith('.ipynb') || resourcePath.fsPath.includes('.ipynb#')) {
                     throw new Error('Unable to access Jupyter kernels for notebook cells');
                 }
-                throw new Error('No environment found');
+                throw new Error('No environment found for the provided resource path: ' + resourcePath.fsPath);
             }
+
+            const execInfo: PythonEnvironmentExecutionInfo = environment.execInfo;
+            const run: PythonCommandRunConfiguration = execInfo.run;
+            envInfo.runCommand = run.executable + (run.args && run.args.length > 0 ? ` ${run.args.join(' ')}` : '');
+            // TODO: check if this is the right way to get type
+            envInfo.type = environment.envId.managerId.split(':')[1];
+            envInfo.version = environment.version;
+
+            // does this need to be refreshed prior to returning to get any new packages?
```

Comment:
Eventually, we should not need to refresh packages. This provides a good fall back if terminal is used to install packages for some reason.

Reply:
will leave for now but add todo